cmd/kaniko-ecr/main.go

```diff
@@ -771,3 +795,26 @@ func isKanikoVersionBelowOneDotEight(v string) bool {
 
 	return currVer.LessThan(oneEightVer)
 }
+
+func getOidcCreds(oidcToken, assumeRole string) (string, string, string) {
+	// Create a new session
+	sess, _ := session.NewSession()
+
+	// Create a new STS client
+	svc := sts.New(sess)
+
+	// Prepare the input parameters for the STS call
+	duration := int64(time.Hour / time.Second)
+	input := &sts.AssumeRoleWithWebIdentityInput{
+		RoleArn:          aws.String(assumeRole),
+		RoleSessionName:  aws.String("kaniko-ecr-oidc"),
+		WebIdentityToken: aws.String(oidcToken),
+		DurationSeconds:  aws.Int64(duration),
+	}
+
+	// Call the AssumeRoleWithWebIdentity function
+	result, _ := svc.AssumeRoleWithWebIdentity(input)
+
```

Comment:
can we return error if **svc.AssumeRoleWithWebIdentity(input)** return error . lets add basic error handling

Reply:
added